nwbinspector/utils.py

```diff
@@ -0,0 +1,30 @@
+"""Authors: Cody Baker and Ben Dichter."""
+import numpy as np
+from collections import defaultdict
+
+global available_checks
+available_checks = {1: defaultdict(list), 2: defaultdict(list), 3: defaultdict(list)}
+
+
+def nwbinspector_check(severity: int, neurodata_type):
```

Comment:
@bendichter The `neurodata_type` set by the wrapper could also be optionally inferred directly from inspecting the signature of the `check_function` (so long as it utilized annotation typing, which is good practice to have when possible anyway); what do you think?

Reply:
I think that's fine as long as the use of typing is made a requirement, i.e., I would have the decorator raise an exception if no type information is given and mention the requirement for use of typing in the docstring of the decorator. The only slight disadvantage is that many users of Python may not yet be familiar with the typing feature and when having this as an explicit required parameter the error can be detected earlier (e.g., though static checks) whereas when using typing errors will appear at runtime (since there is no way to statically enforce typing in Python). There is also slight semantic different, ``neurodata_type`` refers to a type in the schema whereas the type you would use for typing in the function signature is a container class. This difference is probably not critical here, I just wanted to mention just in case it is. Long story short, since it is a required input to the decorator, I think there is some advantage to having it as an explicit input as it makes the API a bit more approachable for others, but enforcing the use of typing is not a bad thing so if you prefer to do it that way, I think its fine.